init/cmdline_test.go

```diff
@@ -50,63 +50,100 @@ func TestGetNextParam(t *testing.T) {
+		// "param02=test02"
+		{"\"param02=test02\"", "param02", "test02", 16},
+		// param03="test03"
+		{"param03=\"test03\"", "param03", "test03", 16},
+		// '   param04=test04   '
+		{"   param04=test04   ", "param04", "test04", 18},
+		// param05=te\0st05
+		{"param05=te\000st05", "param05", "te", 11},
+		// [tab]param06=test06[tab]
+		{"\tparam06=test06\t", "param06", "test06", 16},
+		// param07=te"st07
+		{"param07=te\"st07", "param07", "te\"st07", 15},
+		// par"am08=test08
+		{"par\"am08=test08", "par\"am08", "test08", 15},
+		// param09=test09=test10
+		{"param09=test09=test10", "param09", "test09=test10", 21},
+		// param10=\"test11=test12\"
+		{"param10=\"test11=test12\"", "param10", "test11=test12", 23},
 		// param11
 		{"param11", "param11", "", 7},
 		// "param12"
 		{"\"param12\"", "param12", "", 9},
 		// param13=
 		{"param13=", "param13", "", 8},
-		// param14=te\ st0
-		{"param14=te\\ st0", "param14", "te st0", 15},
-		// param15="te\"st0"
-		{"param15=\"te\\\"st0\"", "param15", "te\"st0", 17},
-		// param16=te"st0
-		{"param16=te\"st0", "param16", "te\"st0", 14},
-		// param17="te"st0"
-		{"param17=\"te\"st0\"", "param17", "te", 12},
-		// param18"=test0
-		{"param18\"=test0", "param18\"", "test0", 14},
-		// param19=te\nst0
-		{"param19=te\nst0", "param19", "te", 11},
-		// param20=test0\r
-		{"param20=test0\r", "param20", "test0", 14},
-		// =test0 // This is a worst case bad junk input, it will return empty key
-		{"=test0", "", "test0", 6},
-		// param21="test0 param22="test1" // This is a worst case bad junk input, it will mangle 21 and swallow 22
-		{"param21=\"test0 param22=\"test1\"", "param21", "test0 param22=", 24},
-		// param23="test0 param24=test1 // This is a worst case bad junk input, it will mangle 23 and swallow 24
-		{"param23=\"test0 param24=test1", "param23", "test0 param24=test1", 28},
+		// param14=te\ st14
+		{"param14=te\\ st14", "param14", "te st14", 16},
+		// param15="te\"st15"
+		{"param15=\"te\\\"st15\"", "param15", "te\"st15", 18},
+		// param16=te"st16
+		{"param16=te\"st16", "param16", "te\"st16", 15},
+		// param17="te"st17"
+		{"param17=\"te\"st17\"", "param17", "te", 12},
+		// param18"=test18
+		{"param18\"=test18", "param18\"", "test18", 15},
+		// param19=te\nst19
+		{"param19=te\nst19", "param19", "te", 11},
+		// param20=test20\r
+		{"param20=test20\r", "param20", "test20", 15},
+		// =test21 // This is a worst case bad junk input, it will return empty key
+		{"=test21", "", "test21", 7},
+		// param22="test22 param23="test23" // This is a worst case bad junk input, it will mangle 22 and swallow 23
+		{"param22=\"test22 param23=\"test23\"", "param22", "test22 param23=", 25},
+		// param24="test24 param25=test25 // This is a worst case bad junk input, it will mangle 24 and swallow 25
+		{"param24=\"test24 param25=test25", "param24", "test24 param25=test25", 30},
 	}
 
 	for _, test := range tests {
 		var k, v string
 		var i int
 
+		t.Log("Testing ", test.input, "\n")
```

Comment:
no need to do the excessive printing

Reply:
I believe the printing will only happen if the test fails, which makes tracking down the failure a lot easier. On success it shouldn't print anything unless run with verbose mode enabled.

I can remove it though if you would prefer.

Reply:
Did not know about it.

Then the change looks good.